### PR TITLE
refactor: ws receive disconnect exc type

### PR DESCRIPTION
**Pull request by peterschutt**
### Pull Request Checklist

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR
- [ ] Pre-Commit Checks were ran and passed
- [ ] Tests were ran and passed

### Description

Modifies `WebSocket.wrapped_receive()` to raise `WebSocketDisconnect` instead of `WebSocketException` where `receive()` is called after a disconnect message has already been received.

### Close Issue(s)

-

### CODE REVIEW

File: litestar/connection/websocket.py [hunk 1]
```diff
@@ -167,8 +167,6 @@ async def receive_data(self, mode: WebSocketMode) -> str | bytes:
         event = cast("WebSocketReceiveEvent | WebSocketDisconnectEvent", await self.receive())
         if event["type"] == "websocket.disconnect":
             raise WebSocketDisconnect(detail="disconnect event", code=event["code"])
-        if self.connection_state == "disconnect":
-            raise WebSocketDisconnect(detail=DISCONNECT_MESSAGE)  # pragma: no cover
```
Comment (peterschutt) [inline litestar/connection/websocket.py:171]: FWICT - I don't think this code is reachable - which probably explains the pragma.

This method calls `self.receive()` which is really `self.receive_wrapper()`, which will either set `self.connection_state` to `"disconnect"` and return the message if its type is `websocket.disconnect`, or (prior to this PR) raise `WebsocketException` if `self.connection_state == "disconnect"` is already true.

File: litestar/connection/websocket.py [hunk 2]
```diff
@@ -72,7 +72,7 @@ def receive_wrapper(self, receive: Receive) -> Receive:
 
         async def wrapped_receive() -> ReceiveMessage:
             if self.connection_state == "disconnect":
-                raise WebSocketException(detail=DISCONNECT_MESSAGE)
```
Comment (peterschutt) [inline litestar/connection/websocket.py:75]: This is raised when `receive()` is called on the connection object after we have already received a `websocket.disconnect` message. Should we raise `WebSocketDisconnect` as I've done here (which is consistent with the unreachable code that I've removed in `receive_data()`), or leave it as is and continue to raise the base exception?

Comment (JacobCoffee) [reply litestar/connection/websocket.py:75]: Since `WebSocketDisconnect` is a subclass of `WebSocketException` I think its okay?

**Review by JacobCoffee — APPROVED** [review]

File: litestar/connection/websocket.py (hunk 2 above)
Comment (peterschutt) [reply litestar/connection/websocket.py:75]: It's OK for backward compat - but is is correct - i.e., is calling `receive()` on an already closed disconnected connection a different error? Maybe I'm over thinking.

Comment (guacs) [reply litestar/connection/websocket.py:75]: I think it makes sense to raise a `WebSocketDisconnect` here. This way the users can handle disconnection related errors differently from the more generic `WebSocketException` if needed. If the `WebSocketException` is raised, then they would have to check the error code to determine if it was a disconnection or not.

**Review by guacs — APPROVED** [review]

File: litestar/connection/websocket.py (hunk 2 above)
Comment (peterschutt) [reply litestar/connection/websocket.py:75]: OK, will roll with it!

**Pull request merged**